superset/utils/pandas_postprocessing.py

```diff
@@ -0,0 +1,323 @@
+    df: DataFrame,
+    columns: Dict[str, str],
+    rolling_type: str,
+    window: int,
+    center: bool = False,
+    win_type: Optional[str] = None,
+    min_periods: Optional[int] = None,
+) -> DataFrame:
+    """
+    Apply a rolling window on the dataset. See the Pandas docs for further details:
+    https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rolling.html
+
+    :param df: DataFrame on which the rolling period will be based.
+    :param columns: columns on which to perform rolling, mapping source column to
+           target column. For instance, `{'y': 'y'}` will replace the column `y` with
+           the rolling value in `y`, while `{'y': 'y2'}` will add a column `y2` based
+           on rolling values calculated from `y`, leaving the original column `y`
+           unchanged.
+    :param rolling_type: Type of rolling window. Any numpy function will work.
+    :param center: Should the label be at the center of the window.
+    :param win_type: Type of window function.
+    :param window: Size of the window.
+    :param min_periods:
+    :return: DataFrame with the rolling columns
+    :raises ChartDataValidationError: If the request in incorrect
+    """
+    _validate_columns(df, columns.keys())
+    df_rolling = df[columns.keys()]
+    if not hasattr(df_rolling, rolling_type):
+        raise ChartDataValidationError(
+            _("Unsupported rolling_type: %(type)s", type=rolling_type)
+        )
+    kwargs: Dict[str, Union[str, int]] = {}
+    if not window:
+        raise ChartDataValidationError(_("Undefined window for rolling operation"))
+
+    kwargs["window"] = window
+    if min_periods is not None:
+        kwargs["min_periods"] = min_periods
+    if center is not None:
+        kwargs["center"] = center
+    if win_type is not None:
+        kwargs["win_type"] = win_type
+
+    df_rolling = df_rolling.rolling(**kwargs)
+    df_rolling = getattr(df_rolling, rolling_type)()
+    df = _append_columns(df, df_rolling, columns)
+    if min_periods:
+        df = df[min_periods:]
+    return df
+
+
+def select(df: DataFrame, columns: Dict[str, str],) -> DataFrame:
+    """
+    Only select a subset of columns in the original dataset. Can be useful for
+    removing unnecessary intermediate results, renaming and reordering columns.
+
+    :param df: DataFrame on which the rolling period will be based.
+    :param columns: Columns on which to perform dff, mapping the
+                    column name to its alias. For instance, `{'y': 'y'}` will return
+                    a DataFrame with only the contents of column `y`,
+                    while `{'y': 'y2'}` return a DataFrame with the column `y2`
+                    containing the values from column `y`.
+    :return: Subset of columns in original DataFrame
+    :raises ChartDataValidationError: If the request in incorrect
+    """
+    _validate_columns(df, columns.keys())
+    return df[columns.keys()].rename(columns=columns)
```

Comment:
What if I want to select the columns in a determinant order? I'd recommend having two separate arguments. 

```python
def select(
    df: DataFrame,
    columns: List[str],
    rename: Optional[Dict[str, str]] = None
) -> DataFrame:
    _validate_columns(df, columns.keys())
    if rename is not None:
        df = df.rename(columns=columns)
    return df[columns.keys()]
```

Reply:
Separating rename from columns is probably a good idea, as always requiring a `Dict[str, str]` even for selecting (=dropping or repordering columns) isn't really intuitive.

Reply:
Yes, that'd be another reason for the move.